omnibot/processor.py

```diff
@@ -101,6 +113,49 @@ def _process_message_handlers(message):
         _handle_help(message)
 
 
+def _process_reaction_message_handlers(reaction: Reaction):
+    bot = reaction.bot
+    statsd = stats.get_statsd_client()
+    handler_called = False
+    item_channel = reaction.item_channel
+    item_ts = reaction.item_ts
+
+    if not _is_message_from_bot(bot, item_channel, item_ts):
+        statsd.incr("event.ignored")
+        return
+
+    for handler in bot.message_handlers:
+        if handler.get("match_type") == "reaction":
+            match = bool(re.fullmatch(handler["match"], reaction.reaction))
+            regex_should_not_match = handler.get("regex_type") == "absence"
+            # A matched regex should callback only if the regex is supposed to
+            # match. An unmatched regex should callback only if the regex is
+            # not supposed to match.
+            if match != regex_should_not_match:
+                reaction.set_match("regex", handler["match"])
+                for callback in handler["callbacks"]:
+                    _handle_message_callback(reaction, callback)
+                    handler_called = True
+
+    if handler_called:
+        statsd.incr("event.handled")
+    elif not handler_called:
+        logger.debug("no handler found")
+        statsd.incr("event.ignored")
+
+
+def _is_message_from_bot(bot: Bot, channel: str, ts: str):
```

Comment:
comment why this is necessary vs just looking up bot from lookup table of registered handlers